### PR TITLE
Introduce Insights feature

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Introduce Insights feature
```

### CODE REVIEW

File: pkg/config/control_plane.go [hunk 1]
```diff
@@ -195,7 +194,7 @@ type InsightCollectorApplication struct {
 }
 
 type InsightCollectorDeployment struct {
-	Enabled bool `json:"enabled"`
+	Enabled bool `json:"enabled" default:"true"`
```
Comment (khanhtc1202) [inline pkg/config/control_plane.go:197]: should be `*bool` if its default value is true
like this: https://github.com/pipe-cd/pipecd/blob/master/pkg/config/application_kubernetes.go#L96

Comment (nghialv) [reply pkg/config/control_plane.go:197]: Oops. Nice catch!

Comment (nghialv) [reply pkg/config/control_plane.go]: @khanhtc1202 Addressed it!

**Review by khanhtc1202 — APPROVED** [review]
Can't wait to see it on prod 🍠

**Pull request merged**